src/Core/src/Platform/iOS/KeyboardAutoManagerScroll.cs

```diff
@@ -0,0 +1,614 @@
+	static int DebounceCount = 0;
+	static NSObject? WillShowToken = null;
+	static NSObject? DidHideToken = null;
+	static NSObject? TextFieldToken = null;
+	static NSObject? TextViewToken = null;
+
+	// Set up the observers for the keyboard and the UITextField/UITextView
+	internal static void Init()
+	{
+		TextFieldToken = NSNotificationCenter.DefaultCenter.AddObserver(new NSString("UITextFieldTextDidBeginEditingNotification"), async (notification) =>
+		{
+			if (notification.Object is not null)
+			{
+				View = (UIView)notification.Object;
+				RootController = View.FindResponder<ContainerViewController>()?.View;
+
+				await SetUpTextEdit();
+			}
+		});
+
+		TextViewToken = NSNotificationCenter.DefaultCenter.AddObserver(new NSString("UITextViewTextDidBeginEditingNotification"), async (notification) =>
+		{
+			if (notification.Object is not null)
+			{
+				View = (UIView)notification.Object;
+				RootController = View.FindResponder<ContainerViewController>()?.View;
+
+				await SetUpTextEdit();
+			}
+		});
+
+		WillShowToken = NSNotificationCenter.DefaultCenter.AddObserver(new NSString("UIKeyboardWillShowNotification"), async (notification) =>
+		{
+			NSObject? frameSize = null;
+			NSObject? curveSize = null;
+
+			var foundFrameSize = notification.UserInfo?.TryGetValue(new NSString("UIKeyboardFrameEndUserInfoKey"), out frameSize);
+			if (foundFrameSize == true && frameSize is not null)
+			{
+				var frameSizeRect = DescriptionToCGRect(frameSize.Description);
+				if (frameSizeRect is not null)
+					KeyboardFrame = (CGRect)frameSizeRect;
+			}
+
+			var foundAnimationDuration = notification.UserInfo?.TryGetValue(new NSString("UIKeyboardAnimationDurationUserInfoKey"), out curveSize);
+			if (foundAnimationDuration == true && curveSize is not null)
+			{
+				var num = (NSNumber)NSObject.FromObject(curveSize);
+				AnimationDuration = (double)num;
+			}
+
+			if (TopViewBeginOrigin == InvalidPoint && RootController is not null)
+				TopViewBeginOrigin = new CGPoint(RootController.Frame.X, RootController.Frame.Y);
+
+			if (!IsKeyboardShowing)
+			{
+				await AdjustPositionDebounce();
+				IsKeyboardShowing = true;
+			}
+		});
+
+		DidHideToken = NSNotificationCenter.DefaultCenter.AddObserver(new NSString("UIKeyboardWillHideNotification"), (notification) =>
+		{
+			NSObject? curveSize = null;
+
+			var foundAnimationDuration = notification.UserInfo?.TryGetValue(new NSString("UIKeyboardAnimationDurationUserInfoKey"), out curveSize);
+			if (foundAnimationDuration == true && curveSize is not null)
+			{
+				var num = (NSNumber)NSObject.FromObject(curveSize);
+				AnimationDuration = (double)num;
+			}
+
+			if (LastScrollView is not null)
+			{
+				AnimateScroll(() =>
+				{
+					if (LastScrollView.ContentInset != StartingContentInsets)
+					{
+						LastScrollView.ContentInset = StartingContentInsets;
+						LastScrollView.ScrollIndicatorInsets = StartingScrollIndicatorInsets;
+					}
+
+					var superScrollView = LastScrollView;
```

Comment:
This is a closure, is capturing the LastScroolView which is a memory leak.

Reply:
Fixed!